src/commands/deployAzurePipeline/configureCicdPipeline/configurePipeline.ts

```diff
@@ -0,0 +1,47 @@
+import * as vscode from 'vscode';
+import * as k8s from 'vscode-kubernetes-tools-api';
+import { IActionContext } from 'vscode-azureextensionui';
+import { resourceNode } from './models/resourceNode';
+const CONFIGURE_PIPELINE_COMMAND = 'configure-cicd-pipeline';
+const DEPLOY_TO_AZURE_EXTENSION_ID = 'ms-vscode-deploy-azure.azure-deploy';
+
+export async function configurePipeline(context: IActionContext, target: any): Promise<void> {
+    const deployToAzureExtensionInstalled = isDeployToAzureExtensionInstalled();
+    if (deployToAzureExtensionInstalled) {
+        const cloudExplorer = await k8s.extension.cloudExplorer.v1;
+        if (cloudExplorer.available) {
+            const clusterTarget = cloudExplorer.api.resolveCommandTarget(target);
+            if (clusterTarget && clusterTarget.cloudName === "Azure" && clusterTarget.nodeType === "resource" && clusterTarget.cloudResource.nodeType === "cluster") {
+                const cluster = { resource: clusterTarget.cloudResource, subscriptionId: clusterTarget.cloudResource.subscription.subscriptionId };
+                await executeDeployToAzureExtensionInstalled(CONFIGURE_PIPELINE_COMMAND, cluster);
+            } else {
+                vscode.window.showInformationMessage('This command only applies to AKS clusters.');
+            }
+        }
+    } else {
+        vscode.window.showWarningMessage('Please install/enable `Deploy to Azure` extension and start again.');
+        await vscode.commands.executeCommand('extension.open', DEPLOY_TO_AZURE_EXTENSION_ID);
+    }
+}
+
+function isDeployToAzureExtensionInstalled(): boolean {
+    const pipelinesExtension = vscode.extensions.getExtension(DEPLOY_TO_AZURE_EXTENSION_ID);
+    if (!pipelinesExtension) {
+        return false;
+    }
+    return true;
+}
+
+async function executeDeployToAzureExtensionInstalled(commandToRun: string, cluster: resourceNode): Promise<void> {
+    const listOfCommands = await vscode.commands.getCommands(true);
```

Comment:
💡 Can you please double check if `getCommands` is an `async` function? definition here doesn't specify that. https://code.visualstudio.com/api/references/vscode-api

Reply:
getCommands in not an async function
But usage of ``` includes ``` will require an await. WiIl update.

Reply:
💡 I wonder why `await` is added here: `includes` just runs on the `list`.

* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes 
* Typescript Playground for `includes` here: https://www.typescriptlang.org/play?#code/MYewdgzgLgBADgUyhGBeGBtA5MAhlLAGhiwBMQBzIkgI3ywF0BuAWACh3RIQAbBAOh6UAFAAMQAVyhwpMAJYoAJAG9EyfnLDAeE0ggjCc9AJQBfUcfZA